rust/libnewsboat/src/utils.rs

```diff
@@ -1179,4 +1196,15 @@ mod tests {
         let input = expected.to_owned() + "# and a comment";
         assert_eq!(strip_comments(&input), expected);
     }
+
+    #[test]
+    fn t_extract_filter() {
+        let expected = ("~/bin/script.sh", "https://newsboat.org");
+        let input = "filter:~/bin/script.sh:https://newsboat.org";
+        assert_eq!(extract_filter(input), expected);
+
+        let expected = ("~/bin/script.sh", "https://newsboat.org");
+        let input = "filter: ~/bin/script.sh:https://newsboat.org";
```

Comment:
Does this test pass with the original C++ implementation?

Reply:
This was not in the original C++ implementation but it passes the C++ tests.

Reply:
> it passes the C++ tests.

No it doesn't.

1. I checked out current `master`, dd14e2bd1e988b9a1566a2cd2ca30d9aa75d8afe
2. I edited the C++ test:

    ```c++
    diff --git a/test/utils.cpp b/test/utils.cpp
    index e9015e01..4e037bbe 100644
    --- a/test/utils.cpp
    +++ b/test/utils.cpp
    @@ -368,7 +368,7 @@ TEST_CASE("extract_filter()", "[utils]")
            std::string filter;
            std::string url;
     
    -       utils::extract_filter("filter:~/bin/script.sh:https://newsboat.org", filter,
    +       utils::extract_filter("filter: ~/bin/script.sh:https://newsboat.org", filter,
                    url);
     
            REQUIRE(filter == "~/bin/script.sh");
    ```
3. I compiled and ran it:

        $ make -j9 test && cd test && TMPDIR=/dev/shm/ ./test

4. It failed:
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    test is a Catch v2.11.1 host application.
    Run with -? for options

    -------------------------------------------------------------------------------
    extract_filter()
    -------------------------------------------------------------------------------
    test/utils.cpp:366
    ...............................................................................

    test/utils.cpp:374: FAILED:
      REQUIRE( filter == "~/bin/script.sh" )
    with expansion:
      " ~/bin/script.sh" == "~/bin/script.sh"

Reply:
Oh, but the rust script works, should we keep it or throw it? Or is the rust tests even being runned?